pkg/sysinfo/sysinfo_linux.go

```diff
@@ -26,225 +26,241 @@ func findCgroupMountpoints() (map[string]string, error) {
 	return mps, nil
 }
 
+type infoCollector func(info *SysInfo, cgMounts map[string]string) (warnings []string)
```

Comment:
FWIW; not super-happy with this signature, but this was a first iteration to clean up; I might do a follow-up to get rid of the `cgMounts map[string]string` together with some other changes